app/views/boxes/_upload_csv_box.haml

```diff
@@ -0,0 +1,146 @@
+        }
+        
+        if (samples.length == 0) {
+          document.getElementById(`csv-file-error`).classList.remove(`hidden`);
+        } 
+        else {
+          // Create row from template with filename and upload info
+          var template = document.getElementById(`csvFileRow`).content;
+          var fragment = template.cloneNode(true);
+
+          fragment.querySelector(`.uploaded-samples-count`).textContent = `${samples.length} Samples ` ;
+          fragment.querySelector(`.upload-icon`).classList.add(`icon-checkmark`)
+          fragment.querySelector(`.file-name`).textContent = `${f.name}`;
+          
+          // Bind click event to remove button
+          var removeFileRow = function(evt) {
+              evt.currentTarget.upload_info.remove();
+              change_event.target.remove();
+              document.getElementById(`add-box-file`).style.display = 'block';
+          }
+          fragment.querySelector(".remove_file").upload_info = fragment.querySelector(`.upload_info`)
+          fragment.querySelector(".remove_file").addEventListener('click', removeFileRow, false)
+
+          // Add tooltip to display not found batches
+          if (not_found_batches.size > 0) {
+            fragment.querySelector(`.uploaded-samples-count`).innerHTML += `(<span class="dashed-underline">${(not_found_batches.size)} batch${(not_found_batches.size>1?'es':'')}</span> not found) `
+            fragment.querySelector(`.upload-icon`).classList.add(`icon-alert`,`icon-red`)
+            fragment.querySelector(`.items-row-action`).classList.add(`ttip`,`input-required`)
+
+            // Add mouse behaviour to display tooltip
+            var ttext = fragment.querySelector(".ttext")
+            ttext.innerHTML = Array.from(not_found_batches).slice(0,5).join("<br>")
+
+            fragment.querySelector(".not_found_message").tooltip = ttext
+            fragment.querySelector(".not_found_message").addEventListener('click', showToolTip, false )
+          } else {
+            fragment.querySelector(`.upload-icon`).classList.add(`icon-check`)
+            fragment.querySelector(`.ttext`).remove()
+          }
+
+          document.getElementById(`uploaded-files`).append(fragment);
+          document.getElementById(`add-box-file`).style.display = 'none';
+          document.getElementById(`csv-file-error`).classList.add(`hidden`);
+        }
+        
+      }
+    })(f);
+    reader.readAsText(f);
+  }
+
+  // Upon pressing the "Add file" button we will create a file input and click it
+  document.getElementById(`add-box-file`).addEventListener('click', function (e) {
+    var template = document.getElementById(`csvInputFile`).content;
+    var fragment = template.cloneNode(true);
+    var fileInput = fragment.querySelector('.csv_file');
+    document.getElementById(`uploaded-files`).append(fragment)
+    
+    fileInput.addEventListener('change', handleFileSelect, false);
+    fileInput.click();
+  })
```

Comment:
I understand that we can only select one CSV file to upload (not many)... what about rendering the file input directly into the main HTML and skip the `add file` button? We'd avoid this whole block of JS, the template, and could directly call `handleFileSelect`:

```haml
#uploaded-files
  = file_field_tag "box[csv_box]", class: "csv_file", accept: "text/csv", onchange: "handleFileSelect"
```